File: NumberTranslatorApi/Services/NumberTranslatorService.fs
module NumberTranslatorService

open TeensTranslatorService
open TensTranslatorService
open IntExtensions

let Translate translateUniqueNumbersWithRepo getPrefixWithRepo number =
    match number with    
    | n when n |> Between 0 12 -> n |> translateUniqueNumbersWithRepo    
    | n when n |> Between 13 19 -> n |> TranslateTeens getPrefixWithRepo    
    | n when n |> Between 20 99 -> n |> TranslateTens getPrefixWithRepo translateUniqueNumbersWithRepo
    | _ -> $"(Translate) Can't translate the number: {number}"    

File: NumberTranslatorStaticClassesApi/Services/NumberTranslatorService.cs
using System;
using NumberTranslatorStaticClassesApi.ExtensionMethods;
using NumberTranslatorStaticClassesApi.Repositories;

namespace NumberTranslatorStaticClassesApi.Services
{
    public static class NumberTranslatorService
    {
        public static Func<int, string> TranslateUniqueNumbersWithRepo()
        {
            return x => UniqueNumberTranslatorService.TranslateUniqueNumbers(UniqueNumberRepository.Get, x);
        }

        public static Func<int, string> GetPrefixWithRepo()
        {
            return x => PrefixService.GetPrefix(NumberPrefixRepository.Get, TranslateUniqueNumbersWithRepo(), x);
        }

        public static string Translate(int number)
        {

            if (number.Between(0, 12))
                return TranslateUniqueNumbersWithRepo()(number);

            if (number.Between(12, 19))
                return TeensTranslatorService.TranslateTeens(GetPrefixWithRepo(), number);

            if(number.Between(20, 99))
                return TensTranslatorService.TranslateTens(GetPrefixWithRepo(), TranslateUniqueNumbersWithRepo(), number);

            return $"(Translate) Can't translate the number: {number}";
        }
    }
}